ts/src/test/base/test.extend.ts

```diff
@@ -0,0 +1,73 @@
+// @ts-nocheck
+// AUTO_TRANSPILE_ENABLED
+
+import assert from 'assert';
+import ccxt from '../../../ccxt.js';
+
+function testBaseExtend () {
+
+    const exchange = new ccxt.Exchange ({
+        'id': 'regirock',
+    });
+
+    const obj1 = {
+        "a": 1,
+        "b": [ 1, 2 ],
+        "c": [ { "test1": 1, "test2": 1 } ],
+        "d": undefined,
+        "e": "not_undefined",
+        "sub": {
+            "a": 1,
+            "b": [ 1, 2 ],
+            "c": [ { "test1": 1, "test2": 2 } ],
+            "d": undefined,
+            "e": "not_undefined",
+            "other1": "x",
+        },
+        "other1": "x",
+    };
+
+    const obj2 = {
+        "a": 2,
+        "b": [ 3, 4 ],
+        "c": [ { "test1": 2, "test3": 3 } ],
+        "d": "not_undefined",
+        "e": undefined,
+        "sub": {
+            "a": 2,
+            "b": [ 3, 4 ],
+            "c": [ { "test1": 2, "test3": 3 } ],
+            "d": "not_undefined",
+            "e": undefined,
+            "other2": "y",
+        },
+        "other2": "y",
+    };
+
+    // extend
+    const extended = exchange.extend (obj1, obj2);
+    tbfeCheckExtended (extended, true);
+    // deepExtend
+    const deepExtended = exchange.deepExtend (obj1, obj2);
+    tbfeCheckExtended (extended, true);
```

Comment:
you're not even using the `deeoExtended` variable here,

this test is not good, the assert function signature depends on the test object (sub property) and the comparison is confusing and we don't even assert that `sub` value is correct, we have to find a better way of comparing these objects

Reply:
ok, this "test.extend" test was actually the "first transpilable" base test as I have mentioned this previously too, we can totally ignore the accuracy of tests inside this file, because the scope of this PR is more important, and we can make the deep/extend's tests better anytime after merge

Reply:
i mean i wrote the content of this test without much in-advance planning, so i didn't want we delay the PR for this one specific test, but if you think we should find a better solution at first, ok, i might also need your thought how to make this better at this moment

Reply:
ok let's ignore this test for now